src/EditorFeatures/VisualBasic/LineCommit/CommitFormatter.vb

```diff
@@ -68,31 +73,32 @@ Namespace Microsoft.CodeAnalysis.Editor.VisualBasic.LineCommit
                 End If
 
                 ' create commit formatting cleanup provider that has line commit specific behavior
+                Dim tree = If(blocking,
+                    document.GetSyntaxTreeSynchronously(cancellationToken),
+                    Await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(False))
                 Dim commitFormattingCleanup = GetCommitFormattingCleanupProvider(
-                                                document,
-                                                documentOptions,
-                                                spanToFormat,
-                                                baseSnapshot, baseTree,
-                                                dirtyRegion, document.GetSyntaxTreeSynchronously(cancellationToken),
-                                                cancellationToken)
+                    document,
+                    documentOptions,
+                    spanToFormat,
+                    baseSnapshot, baseTree,
+                    dirtyRegion, tree,
+                    cancellationToken)
 
                 Dim codeCleanups = CodeCleaner.GetDefaultProviders(document).
                                                WhereAsArray(s_codeCleanupPredicate).
                                                Concat(commitFormattingCleanup)
 
                 Dim finalDocument As Document
                 If useSemantics OrElse isExplicitFormat Then
-                    finalDocument = CodeCleaner.CleanupAsync(document,
-                                                             textSpanToFormat,
-                                                             codeCleanups,
-                                                             cancellationToken).WaitAndGetResult(cancellationToken)
+                    finalDocument = Await CodeCleaner.CleanupAsync(
+                        document, textSpanToFormat, codeCleanups, cancellationToken).ConfigureAwait(False)
```

Comment:
no way to do anything special here if 'blocking' is true. a .wait call here is equivalent to a .Wait higher up in the stack.